### ISSUE TITLE
Improvements for nf-core sync --all

### ISSUE THREAD

**Issue by ewels** [nf-core/tools#548]
Now that we finally got to see `nf-core sync --all` [run for real with all repos](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true), we can see a few improvements that would be good:

* [x] Configure `git` so that commits are by @nf-core-bot
* [x] Skip archived repos, eg. `exoseq`
* [ ] Parse errors from GitHub when we can for nicer output
  * eg. Print just `errors[0]['message']` from problems [like this one](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:223)
  * Once we have nice error messages, can group the final summary by error msg / type
* [ ] Figure out why some pipelines [got `404` errors](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:100)
* [x] Less logging, the `INFO` statements when a pipeline starts isn't really helpful and just crowds the log
* [x] Don't double-log the error log messages
* [ ] ~~Drop the double-line spacing for log messages during the sync~~ _Too much faff_
* [x] Better colours, give some green for successful PRs
* [ ] Print verbose log to a file and `cat` that in a subsequent GitHub Actions step for detailed debugging

Successfully synchronised 22 pipelines 🎉 

Errors whilst synchronising 17 pipelines:

 - [bactmap](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:29) - no `TEMPLATE` branch
 - [clinvap](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:47) - **archived** (no `TEMPLATE` branch, but shouldn't have run)
 - [crisprvar](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:57) - A pull request already exists for `nf-core:TEMPLATE`.
 - [ddamsproteomics](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:75) - A pull request already exists for `nf-core:TEMPLATE`.
 - [deepvariant](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:97) - **404 error**
 - [denovohybrid](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:119) - The `TEMPLATE` branch has no history in common with dev
 - [diaproteomics](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:135) - no `TEMPLATE` branch
 - [exoseq](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:149) - **archived** (Could not parse config options needed for `TEMPLATE`)
 - [hic](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:171) - The `TEMPLATE` branch has no history in common with dev
 - [hlatyping](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:187) - Could not parse config options needed for `TEMPLATE`
 - [kmermaid](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:196) - A pull request already exists for `nf-core:TEMPLATE`.
 - [lncpipe](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:215) - A pull request already exists for `nf-core:TEMPLATE`.
 - [mag](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:237) - **404 error**
 - [mhcquant](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:256) - **404 error**
 - [nascent](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:279) - **404 error**
 - [slamseq](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:312) - no `TEMPLATE` branch
 - [vipr](https://github.com/nf-core/tools/runs/458082607?check_suite_focus=true#step:6:326) - Could not parse config options needed for `TEMPLATE`

So in summary, ignoring archived pipelines:

* 4 x 404 errors
* 4 x "pull request already open" errors
* 2 x "missing config options" errors
* 3 x "missing `TEMPLATE` branch" errors
* 2 x "no history in common with `dev`" errors

The 404 error is the one I find most concerning, as I don't understand why that would happen. The full error is also not very helpful:

`GitHub API returned code 404: `
```json
{
    "message": "Not Found",
    "documentation_url": "https://developer.github.com/v3/pulls/#create-a-pull-request"
}
```

**Comment by ewels** [comment]
@nservant and @caspargross - it looks like you guys both have `TEMPLATE` branches in your repositories which have just been updated by the automated sync, however they have no shared history which means that the PR cannot be created automatically.

In order to fix this for future pipeline syncs, you need to do one fully manual merge. The method for this is documented here: https://nf-co.re/developers/sync#merge-template-into-main-branches
The key is basically to supply `--allow-unrelated-histories` when running the merge.

You'll need to pull the template changes to your fork first, see [these docs](https://nf-co.re/developers/sync#pull-the-changes-to-your-fork).

Once you've done this once, there will be shared history, and future automated sync PRs should work..

Phil

**Comment by ewels** [comment]
Ah maaaaaan

![image](https://user-images.githubusercontent.com/465550/87320401-2ef07e80-c52b-11ea-9b05-66cfff108a15.png)

**Comment by ewels** [comment]
Archived repos should now be skipped thanks to https://github.com/nf-core/tools/pull/671

**Comment by ewels** [comment]
Closing for now. I'm sure that we will want to revisit some of this after the next sync, but now that the logs are dead it's tricky to do more.

TODO for next time: copy relevant log lines into the issue 😅
